external/vcm/vcm/data_transform.py

```diff
@@ -0,0 +1,117 @@
+import dataclasses
+from toolz.functoolz import curry
+from typing import Any, Callable, Literal, Mapping, MutableMapping, Sequence, Set
+import xarray as xr
+import vcm
+
+
+DATA_TRANSFORM_REGISTRY: MutableMapping[str, Mapping[str, Any]] = {}
+
+TransformName = Literal[
+    "Q1_from_Qm_Q2",
+    "Qm_from_Q1_Q2",
+    "Q1_from_Qm_Q2_temperature_dependent",
+    "Qm_from_Q1_Q2_temperature_dependent",
+    "Q1_from_dQ1_pQ1",
+    "Q2_from_dQ2_pQ2",
+]
+
+
+@curry
+def register(
+    inputs: Sequence[str], outputs: Sequence[str], func: Callable[..., xr.Dataset],
+):
+    name = func.__name__
+    if name in DATA_TRANSFORM_REGISTRY:
+        raise ValueError(f"Function {name} has already been added to registry.")
+    DATA_TRANSFORM_REGISTRY[name] = {
+        "func": func,
+        "inputs": inputs,
+        "outputs": outputs,
```

Comment:
optional: you could add more type-safety here with a dataclass instead of Mapping[str, Any].

Reply:
good idea!